code/game/objects/items/devices/healthanalyzerpro.dm

```diff
@@ -0,0 +1,425 @@
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throwforce = 5
+	w_class = W_CLASS_MEDIUM
+	throw_speed = 4
+	starting_materials = list(MAT_IRON = 700, MAT_PLASTIC = 200, MAT_URANIUM = 50, MAT_SILVER = 50, MAT_GOLD = 50)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = Tc_MAGNETS + "=4;" + Tc_BIOTECH + "=4"
+	attack_delay = 0
+	var/last_scantime = 0
+	var/last_reading = null
+	var/mode = "Health Scan"
+	var/list/modes = list("Health Scan", "Simplified Health Scan", "Advanced Health Scan", "Reagents Scan", "Immunity Scan", "Autopsy Scan")
+	var/obj/item/device/antibody_scanner/immune
+	var/last_print
+	parent_type = /obj/item/weapon/autopsy_scanner
+
+/obj/item/device/healthanalyzerpro/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Current active mode: [mode].</span>")
+
+/obj/item/device/healthanalyzerpro/verb/toggle_mode()
+	set name = "Switch mode"
+	set category = "Object"
+
+	mode = input(usr, "Please select module.", "Pro Health Scanner") in modes
+	last_reading = null
+	last_scantime = 0
+
+/obj/item/device/healthanalyzerpro/verb/print(mob/user)
+	set name = "Print from memory"
+	set category = "Object"
+
+	if(mode == "Immunity Scan")
+		to_chat(user, "<span class='warning'>Due to memory constraints, immunity scan doesn't provide printing function!</span>")
+		return
+	if(!last_reading)
+		to_chat(user, "<span class='warning'>The memory is empty.</span>")
+		return
+	if(!user.dexterity_check()) //it's a complex thingy
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return
+	if(last_print + 150 > world.time)
+		to_chat(user, "<span class='warning'>Printing energy spent, please wait a moment.</span>")
+		return
+
+	for(var/mob/O in viewers(usr))
+		O.show_message("<span class='warning'>\the [src] rattles and prints out a sheet of paper.</span>", 1)
+	sleep(10)
+	last_print = world.time
+	var/obj/item/weapon/paper/R = new(loc)
+	R.name = "paper - '[mode] results'"
+	R.info = last_reading
+	if(istype(user,/mob/living/carbon))
+		user.put_in_hands(R)
+
+/obj/item/device/healthanalyzerpro/attack(mob/living/L, mob/living/user)
+	if(!user.dexterity_check()) //it's a complex thingy
+		to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+		return
+	if(user.hallucinating())
+		hallucinate_scan(L,user)
+		return
+	switch(mode)
+		if("Health Scan", "Simplified Health Scan")
+			health_scan(L,user)
+		if("Advanced Health Scan")
+			if(istype(L,/mob/living/carbon/human/))
+				body_scan(L,user)
+		if("Autopsy Scan")
+			if(istype(L,/mob/living/carbon/human/))
+				autopsy_scan(L,user)
+	src.add_fingerprint(user)
+
+/obj/item/device/healthanalyzerpro/preattack(atom/O, mob/user) //snowlakes
+	switch(mode)
+		if("Reagents Scan")
+			reagent_scan(O,user)
+		if("Immunity Scan")
+			immune_scan(O,user)
+	src.add_fingerprint(user)
+
+/obj/item/device/healthanalyzerpro/attack_self(mob/living/user)
+	. = ..()
+	if(.)
```

Comment:
delete last two lines, substitute if(..()) instead

Reply:
That's not equivalent. The current method should be used if you want trues to propagate.

Reply:
I don't think it's necessary for attack_self though (as opposed to preattack or attack)

Reply:
Probably not necessary, but in that case either way is valid